### PR TITLE
VMware: Refactor vmware_maintenancemode module

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/vmware/vmware_maintenancemode.py

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

File: lib/ansible/modules/cloud/vmware/vmware_maintenancemode.py [hunk 1]
```diff
@@ -97,109 +101,96 @@
...
+            spec.vsanMode.objectAction = self.vsan
+
+        try:
+            task = self.host.EnterMaintenanceMode_Task(self.module.params['timeout'],
+                                                       self.module.params['evacuate'],
+                                                       spec)
+
+            success, result = wait_for_task(task)
+
+            return self.module.exit_json(changed=success,
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_maintenancemode.py]: I think the `return` isn't necessary (you don't use it in ExitMaintenanceMode).

**Review by tchernomax — CHANGES_REQUESTED** [review]
A minor modification suggested, else it looks good to me.

File: lib/ansible/modules/cloud/vmware/vmware_maintenancemode.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_maintenancemode.py]: My bad, missed it.

**Review by tchernomax — APPROVED** [review]
shipit

**Pull request merged**